api/config/services.yaml

```diff
@@ -33,3 +35,9 @@ services:
     App\Handler\BookHandler:
         arguments:
             $imgflipClient: '@csa_guzzle.client.imgflip'
+
+    # Custom data provider 
+    App\Repository\TopBook\TopBookCachedDataRepository:
+        decorates: App\Repository\TopBook\TopBookDataRepository
+
+    App\Repository\TopBook\TopBookDataInterface: '@App\Repository\TopBook\TopBookCachedDataRepository'
```

Comment:
```suggestion
    App\Repository\TopBook\TopBookDataInterface: '@App\Repository\TopBook\TopBookDataRepository'
```

Reply:
If we do this, the cached version won't be used and should be deleted, the related decorator and related changes.

Reply:
Nop, cause the TopBookCachedDataRepository decorates the TopBookDataRepository. It'll be used transparently